### PR TITLE
fixes #784

### LINKED ISSUE #784: Home in visual line selection doesn't move cursor.
------

Please *thumbs-up* 👍 this issue if it personally affects you! You can do this by clicking on the emoji-face on the top right of this post. Issues with more thumbs-up will be prioritized.

-----

### What did you do?

```
1
2
[3]
4
5
```

`VH`

### What did you expect to happen?

Lines 1-3 are selected, cursor should move to 1.

### What happened instead?

Lines 1-3 are selected, but cursor stays on line 3.

### Technical details:

Works as expected with `VL`. Also this changes the line we've started selection from to line 1, but this should still be the 3rd line.

### PR DESCRIPTION
**Pull request by xconverge**
Yay! We love PRs! 🎊

Please include a description of your change and ensure:

- [X] Commit message has a short title & issue references
- [X] Each commit does a logical chunk of work.
- [X] It builds and tests pass (e.g `gulp`)

More info can be found on our [contribution guide](https://github.com/VSCodeVim/Vim/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: src/actions/actions.ts [hunk 1]
```diff
@@ -2599,10 +2599,19 @@ abstract class MoveByScreenLine extends BaseMovement {
...
-      return {
-        start: Position.FromVSCodePosition(vscode.window.activeTextEditor.selection.start),
-        stop: Position.FromVSCodePosition(vscode.window.activeTextEditor.selection.end)
-      };
+
+      let start = Position.FromVSCodePosition(vscode.window.activeTextEditor.selection.start);
+      let stop = Position.FromVSCodePosition(vscode.window.activeTextEditor.selection.end);
+
+      // We want to swap the cursor start stop positions based on which direction we are moving, up or down
+      if (start.line < position.line) {
```
Comment (johnfn) [inline src/actions/actions.ts]: You can do a swap like this: `[a, b] = [b, a]`

You could also write this whole block like this:

`return { start: Position.earlierOf(start, stop), stop: Position.laterOf(start, stop) }`

**Comment by johnfn** [comment]
Ideally we would refactor stuff to use `Range` now though...

File: src/actions/actions.ts (hunk 1 above)
Comment (xconverge) [inline src/actions/actions.ts]: Aye, I will change it, this is where all readability goes away in my opinion. I can solve this by just not reading it again :)

**Comment by xconverge** [comment]
Good to go I think

**Comment by johnfn** [comment]
Thanks cubed, @xconverge!!!

**Pull request merged**